src/ristretto/bulletproofs_plus.rs

```diff
@@ -237,9 +237,12 @@ impl RangeProofService for BulletproofsPlusService {
                     minimum_value_promises: vec![None],
                     seed_nonce: None,
                 };
-                match RistrettoRangeProof::verify_do_not_recover_masks(self.transcript_label, &[statement], &[
-                    rp.clone()
-                ]) {
+                match RistrettoRangeProof::verify_batch(
```

Comment:
I'm not a fan of returning different data based on the action, but I think that's subjective, so think it's fine